admin-ui/plugins/admin/components/MAU/MauGraph.js

```diff
@@ -9,35 +9,49 @@ import GluuViewWrapper from 'Routes/Apps/Gluu/GluuViewWrapper'
 import { getMau } from 'Plugins/admin/redux/features/mauSlice'
 import GluuLabel from 'Routes/Apps/Gluu/GluuLabel'
 import { Button, Card, CardFooter, CardBody, FormGroup, Col, Row } from 'Components'
-import { hasBoth, buildPayload, STAT_READ, STAT_JANS_READ } from 'Utils/PermChecker'
+import { buildPayload, STAT_READ, STAT_JANS_READ } from 'Utils/PermChecker'
 import { useTranslation } from 'react-i18next'
 import { useDispatch, useSelector } from 'react-redux'
 import SetTitle from 'Utils/SetTitle'
 import applicationStyle from 'Routes/Apps/Gluu/styles/applicationstyle'
 import { ThemeContext } from 'Context/theme/themeContext'
 import dayjs from 'dayjs'
+import { useCedarling } from '@/cedarling'
 
 const ActiveUsersGraph = lazy(() => import('Routes/Dashboards/Grapths/ActiveUsersGraph'))
 
 function MauGraph() {
   const dispatch = useDispatch()
   const statData = useSelector((state) => state.mauReducer.stat)
   const loading = useSelector((state) => state.mauReducer.loading)
-  const permissions = useSelector((state) => state.authReducer.permissions)
+  const { permissions: cedarPermissions } = useSelector((state) => state.cedarPermissions)
   const { t } = useTranslation()
   const theme = useContext(ThemeContext)
   const selectedTheme = theme.state.theme
   const [startDate, setStartDate] = useState(dayjs().subtract(3, 'months'))
   const [endDate, setEndDate] = useState(dayjs())
+  const { hasCedarPermission, authorize } = useCedarling()
   const userAction = {}
   const options = {}
+  const initPermissions = async () => {
+    const permissions = [STAT_READ, STAT_JANS_READ]
+    for (const permission of permissions) {
+      await authorize([permission])
+    }
+  }
```

Comment:
Authorize both permissions in parallel and track dependency.

Run both authorizations concurrently and include authorize in the effect dependency to avoid stale closures.


Apply:

```diff
-  const initPermissions = async () => {
-    const permissions = [STAT_READ, STAT_JANS_READ]
-    for (const permission of permissions) {
-      await authorize([permission])
-    }
-  }
+  const initPermissions = async () => {
+    await Promise.all([STAT_READ, STAT_JANS_READ].map((p) => authorize([p])))
+  }
```

And update the effect deps:

```diff
-  }, [])
+  }, [authorize])
```